### PR TITLE
Remove deprecated hook code from plugins

### PR DESCRIPTION
**Pull request by kaxil**
We had removed registering hooks via plugin in https://github.com/apache/airflow/pull/12108 for Airflow 2.0. We kept it for registering Connection form.

As I understand it, we don't use that code but `ProvidersManager` (`airflow/providers_manager.py`) to register Connections from providers.

[Our docs](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#plugins) already mark this as removed in 2.0:

image

https://github.com/apache/airflow/blob/156878806618145430629a4069911d425d6bf77a/docs/apache-airflow/authoring-and-scheduling/plugins.rst#L33-L37

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by kaxil** [comment]
@ashb since you were the author of https://github.com/apache/airflow/pull/12108 , could you review this PR and see if it makes sense

**Review by jedcunningham — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

File: tests/plugins/test_plugin.py [hunk 1]
```diff
@@ -136,7 +130,6 @@ class AirflowTestPlugin(AirflowPlugin):
     name = "test_plugin"
     operators = [PluginOperator]
     sensors = [PluginSensorOperator]
-    hooks = [PluginHook]
```
Comment (potiuk) [inline tests/plugins/test_plugin.py:118]: Should we also remove operators/sensors above?

Comment (kaxil) [reply tests/plugins/test_plugin.py:118]: Yup that was removed in https://github.com/apache/airflow/pull/43290

**Comment by kaxil** [comment]
> Should be fine. The plugins from Airlfow 2 will stop working anyway and people will have to modify them. This is pretty much a "given" incompatibility and we need to bite the bullet. But the test needs to be fixed and I think we should add a newsfragment (just one liner now and we can expand it later).

That was already part of the docs though, check PR description -- should we still add a newsfragment:


[Our docs](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#plugins) already mark this as removed in 2.0:


image

https://github.com/apache/airflow/blob/156878806618145430629a4069911d425d6bf77a/docs/apache-airflow/authoring-and-scheduling/plugins.rst#L33-L37

**Review by ashb — APPROVED** [review]
Yeah looks good to me.

I think an explicit news fragment is a good idea, even though the docs say it was deprecated this should be called out as finally breaking. (I'm thinking to the issue FAB/WTForms had recently, so lets be explicit when we can)

**Comment by kaxil** [comment]
Added in https://github.com/apache/airflow/pull/43291/commits/3f92bf7da18a423545bb49d7282a061b1acf3760

**Pull request merged**